internal/hashing/writer.go

```diff
@@ -15,13 +15,14 @@ type Writer struct {
 func NewWriter(w io.Writer, h hash.Hash) *Writer {
 	return &Writer{
 		h: h,
-		w: io.MultiWriter(w, h),
+		w: w,
 	}
 }
 
 // Write wraps the write method of the underlying writer and also hashes all data.
 func (h *Writer) Write(p []byte) (int, error) {
 	n, err := h.w.Write(p)
+	h.h.Write(p[:n])
```

Comment:
The Hash interface states that a call to `Write()` never returns an error. Does this also apply to the number of written bytes?

Reply:
I'm not exactly sure if this is what you mean, but io.Writer "must return a non-nil error if it returns n < len(p)". So if a hash.Writer never returns an error, it must always return the length of its argument.

Reply:
That's exactly the bit of information I've missed. Together with "hash.Write" never returns an error this implies that all n bytes are always written.